src/main/java/org/jenkinsci/plugins/workflow/steps/SetBuildInfoStep.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.steps;
+
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.util.ListBoxModel;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public final class SetBuildInfoStep extends Step {
+
+    private final RunPropertySetter setting;
+
+    @DataBoundConstructor
+    public SetBuildInfoStep(RunPropertySetter setting) {
+        this.setting = setting;
+    }
+
+    public RunPropertySetter getSetting() {
+        return setting;
+    }
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new Execution(this, context);
+    }
+
+    private static class Execution extends SynchronousStepExecution<Void> {
+        private static final long serialVersionUID = 1;
+
+        private transient final SetBuildInfoStep step;
+
+        Execution(SetBuildInfoStep step, StepContext context) {
+            super(context);
+            this.step = step;
+        }
+
+        @Override
+        public Void run() throws Exception {
+            Run r = getContext().get(Run.class);
+            if (r != null) {
```

Comment:
It is required context; you need not and should not check for null.

Reply:
IDEs tend to glare at you if you don't. =)